Framework/LiveData/src/Kafka/KafkaEventStreamDecoder.cpp

```diff
@@ -586,16 +586,19 @@ void KafkaEventStreamDecoder::initLocalCaches(
         spDetMsg->spectrum()->data(), spDetMsg->detector_id()->data(), nudet);
   }
 
-  // Load the instrument if possible but continue if we can't
-  if (!instName.empty()) {
-    loadInstrument<DataObjects::EventWorkspace>(instName, eventBuffer,
-                                                jsonGeometry);
-    if (rawMsgBuffer.empty()) {
-      eventBuffer->rebuildSpectraMapping();
+  auto instrument = eventBuffer->getInstrument();
+  if (instrument->getNumberDetectors() != eventBuffer->getNumberHistograms()) {
```

Comment:
I assume this is checking if the instrument in `eventBuffer` is the correct one or the data?
Is number of detectors alone a good enough way to do this?